### ISSUE TITLE
DateTime Format

### ISSUE THREAD

**Issue by BenKoth** [Azure/azure-cosmos-table-dotnet#14]
SDK Version: 1.0.1
Steps to reproduce:

I have a table entity class with a DateTime property. When I query the resulting data using the Azure Document Client, the JSON returned is in a format I don't recognize. The reason I'm using the document client was to try and reproduce how the data is imported to Power BI Desktop. The DateTime appears correct when viewed in the portal or through the Microsoft Azure Storage Explorer.

Here is the resulting JSON value:
```
{
  "id": "tvmps_050e7bf8eedbaf55c4514c1a8d171f4d8e4b6ef9adbc4eb880abe569f401391c_p",
  "RunId": {
    "$t": 2,
    "$v": "e1c2345a-f638-455c-8970-a83b35dc931b"
  },
  "InstanceId": {
    "$t": 2,
    "$v": "tvmps_050e7bf8eedbaf55c4514c1a8d171f4d8e4b6ef9adbc4eb880abe569f401391c_p"
  },
  "StartTime": {
    "$t": 9,
    "$v": "00636988010372592895"
  },
  "EndTime": {
    "$t": 9,
    "$v": "00636988013347507596"
  },
  "MessagesProcessed": {
    "$t": 16,
    "$v": 0
  },
  "IsCompareInstance": {
    "$t": 8,
    "$v": false
  },
  "$pk": "e1c2345a-f638-455c-8970-a83b35dc931b",
  "$id": "tvmps_050e7bf8eedbaf55c4514c1a8d171f4d8e4b6ef9adbc4eb880abe569f401391c_p",
  "_rid": "uFV6ALUqhzkKAAAAAAAAAA==",
  "_self": "dbs/uFV6AA==/colls/uFV6ALUqhzk=/docs/uFV6ALUqhzkKAAAAAAAAAA==/",
  "_etag": "\"00000000-0000-0000-3b22-125cb40401d5\"",
  "_attachments": "attachments/",
  "_ts": 1563204560
}
```

The "StartTime" and "EndTime" fields are the ones I don't know what format it is. The documentation claims DateTime are stored as ISO-8601 format dates, which is how they appear in the portal/storage explorer, but that is not how they appear to be stored in the underlying JSON.

When I look at it in the Azure Storage Explorer it shows up like "2019-07-15T15:23:38.804Z" which looks like the correct ISO-8601 format. 

I'm trying to determine what that value is from the document query so I can convert the date to a readable date in Power BI.

**Comment by donghexu** [comment]
Hi @BenKoth, at the backend the data is stored in the format that is not visible to customers. SDK will do the conversion so when you use Table SDK, you will see the DataTime type property is in the correct format. The same thing to Storage Explorer, there is a translation logic underline. We encourage to use Table SDK to connect with Cosmos DB Table API. 

Thanks,

**Comment by BenKoth** [comment]
That's great except that the Azure Cosmos DB is now an option in Power BI. How are we supposed to translate that value to a DateTime in PowerBI?

**Comment by BenKoth** [comment]
Also, perhaps the documentation should be updated.
https://docs.microsoft.com/en-us/azure/cosmos-db/working-with-dates
That claims the date's are stored in ISO-8601 format when you are saying they are stored in some other format.

**Comment by sakash279** [comment]
@BenKoth can you share the code for how you are serializing and de-serializing your date time object? 

From your use case, it seems you are inserting data using Tables SDK and retrieving it using Document Client?

**Comment by BenKoth** [comment]
I am inserting using the Tables SDK, I am trying to retrieve it into Power BI Desktop, which you have to use the document client for. I had a separate ticket on this on the power bi community as well and they at least gave me a way to format the string. I feel like Power BI needs to update to allow importing dates better since they allow cosmos db as an input source now but also wanted to note the issue here so that the owners of the Cosmos DotNet SDK could also hopefully work with PowerBI better.

https://community.powerbi.com/t5/Issues/Support-DateTime-Populated-Through-Azure-Cosmos-Table-SDK/idi-p/789453#M48323

**Comment by sakash279** [comment]
This is typically an unsupported scenario. i.e. write using Tables SDK and read using DocumentClient (Power BI eventually). PowerBI needs to integrate and consume Tables SDK, which will allow such scenarios to be seamless for the users needs. 

Given you are unblocked, I am going to close the issue for now.
